src/storage.cc

```diff
@@ -0,0 +1,386 @@
+                                  << " just received an order"
+                                  << " for " << it->amt
+                                  << " of " << commodity;
+  }
+
+}
+
+//- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+void Storage::AddMat_(cyclus::Material::Ptr mat) {
+  // Here we do not check that the recipe matches the input recipe. 
+
+  LOG(cyclus::LEV_INFO5, "ComCnv") << prototype() << " is initially holding "
+                                << inventory.quantity() << " total.";
+
+  try {
+    inventory.Push(mat);
+  } catch (cyclus::Error& e) {
+    e.msg(Agent::InformErrorMsg(e.msg()));
+    throw e;
+  }
+
+  LOG(cyclus::LEV_INFO5, "ComCnv") << prototype() << " added " << mat->quantity()
+                                << " of " << in_commod_()
+                                << " to its inventory, which is holding "
+                                << inventory.quantity() << " total.";
+
+}
+
+//- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+cyclus::Material::Ptr Storage::Request_() {
+  double qty = std::max(0.0, current_capacity());
+  LOG(cyclus::LEV_INFO5, "ComCnv") << prototype()
+                                  << " just requested "
+                                  << current_capacity()
+                                  << " of commodity: " << in_commod
+                                  << " with recipe: " << in_recipe;
+  return cyclus::Material::CreateUntracked(qty,
+                                        context()->GetRecipe(in_recipe));
+}
+
+//- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+cyclus::BidPortfolio<cyclus::Material>::Ptr Storage::GetBids_(
+    cyclus::CommodMap<cyclus::Material>::type& commod_requests,
+    std::string commod,
+    cyclus::toolkit::ResourceBuff* buffer) {
+  using cyclus::Bid;
+  using cyclus::BidPortfolio;
+  using cyclus::CapacityConstraint;
+  using cyclus::Composition;
+  using cyclus::Converter;
+  using cyclus::Material;
+  using cyclus::Request;
+  using cyclus::ResCast;
+  using cyclus::toolkit::ResourceBuff;
+
+  BidPortfolio<Material>::Ptr port(new BidPortfolio<Material>());
+
+  if (commod_requests.count(commod) > 0 && buffer->quantity() > 0) {
+    std::vector<Request<Material>*>& requests = commod_requests.at(commod);
+
+    // get offer composition
+    Material::Ptr back = ResCast<Material>(buffer->Pop(ResourceBuff::BACK));
+    Composition::Ptr comp = back->comp();
+    buffer->Push(back);
+
+    std::vector<Request<Material>*>::iterator it;
+    for (it = requests.begin(); it != requests.end(); ++it) {
+      Request<Material>* req = *it;
+      double qty = std::min(req->target()->quantity(), buffer->quantity());
+      Material::Ptr offer = Material::CreateUntracked(qty, comp);
+      port->AddBid(req, offer, this);
+    }
+
+    //CapacityConstraint<Material> cc(buffer->quantity());
+    //port->AddConstraint(cc);
```

Comment:
Here again, the need or absence of need for a capacity constraint should be confirmed or refuted so that we can either delete these commented lines or reinstate them.